Revised FIP-0036 Draft for Community Consideration (#451)
filecoin-project/FIPs: FIPS/fip-0036.md

Thank you all for your continued insights and collaboration on present and future Filecoin Improvement Proposals.

## Simple Summary
- A Sector Duration Multiplier is introduced for all sectors, including Committed Capacity (CC) sectors and sectors containing storage deals.
- A longer sector will have a higher Quality Adjusted Power than a shorter sector, all things equal.
- The Duration Multiplier is multiplicative on the existing Quality Multiplier incentive (Filecoin Plus incentive). Filecoin Plus already offers up to a 10x multiplier for proving storage of data from verified clients. FIP-0036 introduces an independent Duration Multiplier up to 3.5x. Thus, the *maximum* multiplier available to any sector will increase from 10x to 35x. 

> - Sectors with higher Quality Adjusted Power as a result of the Sector Duration Multiplier and Quality Multiplier will require higher initial pledge collateral.

Author: No - all sectors will have the same PCD regardless of their QAP following FIP-0034

> - The minimum sector duration time will increase from 6 months to 1 year and the maximum sector duration will increase from 1.5 years to 3.5 years. This change to maximum sector duration **will be implemented gradually**: we propose increasing the maximum duration by 6 months every week for 4 weeks, until the maximum duration of 3.5 years is available. 

Reviewer: this change is suggested cuz we dont really have a maximum sector duratiom variable but https://github.com/filecoin-project/builtin-actors/blob/7a65393774aea71f9b3cd19c240bc71623ea5a07/runtime/src/runtime/policy.rs#L114

> - The minimum sector duration time will increase from 6 months to 1 year and the maximum sector duration will increase from 1.5 years to 3.5 years. This change to maximum sector duration **will be implemented gradually**: we propose increasing the maximum duration by 6 months every week for 4 weeks, until the maximum duration of 3.5 years is available. 

Reviewer: Note the FIP currently does not propose to change deal_duration_bounds [180,540] accordingly  - is that intentional?

- CEL and the community will monitor the network and look to increase the maximum sector duration in a future FIP if network conditions merit, pending community-driven support for such measures.  
- The SectorInitialConsensusPledge multiplier will increase from 30% to 50%.
- This policy will apply at Sector Upgrade and Extension.

> ### History of FIP-0036

Reviewer: ```suggestion
### Evolution of FIP-0036
```

This FIP seeks to improve the stability of storage commitments underlying the network and strengthen the Filecoin Economy amidst an uncertain macroeconomic climate, ultimately improving the stability and predictability of Storage Provider (SP) returns. 

> Protocol Labs (PL) and CEL researchers investigated potential protocol improvements in April/May, and, began the [initial FIP discussion](https://github.com/filecoin-project/FIPs/discussions/386) in June. Initial parameter recommendations followed in the first [FIP Draft](https://github.com/filecoin-project/FIPs/commit/861601f7f607617d06bf9dc714d96ea2cb19344e?short_path=2026700#diff-2026700082539835fb781f31b11f9316c352b4f85501e43c2f114079abee5b5e), supported by [CEL modeling](https://www.notion.so/Duration-Changes-FIP-discussion-Analysis-Summary-735ce6685b7946f0a03fc13c3fe271fa). Following SP and community feedback, several parameters were adjusted, investigated, and discussed. Over the course of many months in which the community continuously supplied feedback, questions, and identified points of disagreement/concern, FIP-0036 now exists in its current form, supported by analysis and modeling from both CEL and its broader network.

Reviewer: Do we mean CNL here? if so might worth specify and link to the team page

Author: I think it implies then that they were part of the initial FIP draft. I think we can keep as is.

> Protocol Labs (PL) and CEL researchers investigated potential protocol improvements in April/May, and, began the [initial FIP discussion](https://github.com/filecoin-project/FIPs/discussions/386) in June. Initial parameter recommendations followed in the first [FIP Draft](https://github.com/filecoin-project/FIPs/commit/861601f7f607617d06bf9dc714d96ea2cb19344e?short_path=2026700#diff-2026700082539835fb781f31b11f9316c352b4f85501e43c2f114079abee5b5e), supported by [CEL modeling](https://www.notion.so/Duration-Changes-FIP-discussion-Analysis-Summary-735ce6685b7946f0a03fc13c3fe271fa). Following SP and community feedback, several parameters were adjusted, investigated, and discussed. Over the course of many months in which the community continuously supplied feedback, questions, and identified points of disagreement/concern, FIP-0036 now exists in its current form, supported by analysis and modeling from both CEL and its broader network.

Reviewer: ```suggestion
Protocol Labs (PL) and CEL researchers investigated potential protocol improvements in April/May, and, began the [initial FIP discussion](https://github.com/filecoin-project/FIPs/discussions/386) in June. Initial parameter recommendations followed in the first [FIP Draft](https://github.com/filecoin-project/FIPs/commit/861601f7f607617d06bf9dc714d96ea2cb19344e?short_path=2026700#diff-2026700082539835fb781f31b11f9316c352b4f85501e43c2f114079abee5b5e), supported by [CEL modeling](https://www.notion.so/Duration-Changes-FIP-discussion-Analysis-Summary-735ce6685b7946f0a03fc13c3fe271fa). Following SP, core devs and community feedback, several parameters were adjusted, investigated, and discussed. Over the course of many months in which the community continuously supplied feedback, questions, and identified points of disagreement/concern, FIP-0036 now exists in its current form, supported by analysis and modeling from both CEL and its broader network.
```

The motivation to increase the minimum and maximum sector durations is decreased sector turnover and improved network stability. Given that the network expects to grow, having sectors that expire every 6 months means that the network potentially needs to find new sealing throughput to compensate for the loss of power from expiration. This hinders the network as it continues to scale. 

Another perspective is that block rewards are high now but exponentially decreasing, and these high early rewards should be used to incentivize participation that’s long-term aligned with the network. 6 months is not long-term. In particular it's short over the scale we need stability in the supply dynamics. Increasing the minimum duration from 6 months to 1 year doubles the minimum level of commitment, and smooths out locking dynamics by stretching inflow-outflow over a longer time period, all while impacting relatively few Storage Providers as sector durations for CC and FIL+ are both substantially above the minimum on average. 

> Finally, the new maximum limit of 3.5 years gives SPs the option to express a long-term commitment and bullish view on the network which was previously not possible with the maximum sector duration length of 540 days, and also receive commensurate rewards. 

Reviewer: ```suggestion
Finally, the new maximum limit of 3.5 years gives SPs the option to express a long-term commitment and bullish view on the network which was previously not possible with the maximum sector expiration length of 540 days, and also receive commensurate rewards. 
```

| **SectorDurationMultiplier (SDM)** | **Multiplier for power for storage based on promised sector duration** |

**This SectorDurationMultiplier function proposed is linear with slope 1**. See below for the function proposed. 

> ![alt text](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/0c2ed368-a5a2-4431-bc01-25e099e9afc8/Image_9-1-22_at_12.08_PM.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20220901%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20220901T191326Z&X-Amz-Expires=86400&X-Amz-Signature=33aece2261b638cf8d09ab69188920f018cc28c8bbfd7a6f542ea372c52f6ffa&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Image%25209-1-22%2520at%252012.08%2520PM.jpg%22&x-id=GetObject)

Reviewer: broken link. - would suggest you to add the graph to https://github.com/filecoin-project/FIPs/tree/master/resources under an fip-0036 folder

> The rationale to select this linear slope 1 function is based on a principle that the selected parameters should maximize the effectiveness of the duration incentive, subject to SP’s collateral availability constraints, while taking into account micro and macroeconomic consequences with minimal added implementation complexity. Further analysis/simulation is shown in the analysis brief CEL prepared linked [here](https://pl-strflt.notion.site/Duration-FIP-revisions-7426f344685940409ac513a0ffcccc86) and above. 

Reviewer: would suggest to keep this under https://github.com/filecoin-project/FIPs/tree/master/resources as well

As of end of August 2022, the calculations are approximately: 

```
EpochReward := 20.53 * 5 = 101.2 FIL
NetworkPower := 18.28 * 2^60 Bytes
CirculatingSupply := 348.8 * 10^6 FIL

// Sector quality = 1
StoragePledge := 0.0095 FIL
ConsensusPledge := 0.2843 FIL
PreCommitDeposit := 35 * StoragePledge = 0.3325 FIL
InitialPledge := StoragePledge + ConsensusPledge = 0.2938 FIL

// Sector quality = 35 (values are approx 35x greater than a sector of quality 1)
StoragePledge := 0.3325 FIL
ConsensusPledge := 9.9504 FIL
PreCommitDeposit := StoragePledge = 0.3325 FIL
InitialPledge := StoragePledge + ConsensusPledge = 10.2828 FIL 
```
Per the change proposed by the FIP, the ConsensusPledge is calculated as:
- $SectorInitialConsensusPledge = 0.5 \cdot FILCirculatingSupply \cdot \frac{SectorQAP}{max(BaselineTarget, NetworkQAP)}$

> PCD risk was more pronounced in previous iterations of the FIP when pre-commit deposit was larger then Initial Pledges for lower QAP sectors. This was because the maximum sector power was higher at a 50x multiplier. Given that this revised FIP **caps max sector duration at 3.5 years**, and, therefore, caps the maximum QAP multiplier to **35x**, in most cases, we expect PCD to be less than or equal to the initial pledge for a sector, which means that Storage Providers will have to lock at least this much anyway by the time the sector is proven. We expect this relationship to continue to hold, as FIL supply expands over time with long-term vesting and emissions, and (2) the expected reward per sector falls over time with growth in network capacity and decaying block reward. CEL has produced an accompanying [tool](https://docs.google.com/spreadsheets/d/1mia7LyDSOh3lwAGhz7gTPUc65DQJbhHPyh013QpM4v4/edit#gid=0) with supporting documentation, allowing Storage Providers to estimate their expected FIL-on-FIL return given their projected failure rate to proveCommit sectors. 

Reviewer: ```suggestion
Given that this FIP **caps max sector duration at 3.5 years**, and, therefore, caps the maximum QAP multiplier to **35x**, in most cases, we expect PCD to be less than or equal to the initial pledge for a sector, which means that Storage Providers will have to lock at least this much anyway by the time the sector is proven. We expect this relationship to continue to hold, as FIL supply expands over time with long-term vesting and emissions, and (2) the expected reward per sector falls over time with growth in network capacity and decaying block reward. CEL has produced an accompanying [tool](https://docs.google.com/spreadsheets/d/1mia7LyDSOh3lwAGhz7gTPUc65DQJbhHPyh013QpM4v4/edit#gid=0) with supporting documentation, allowing Storage Providers to estimate their expected FIL-on-FIL return given their projected failure rate to proveCommit sectors. 
```

## Backwards Compatibility

> This policy would apply at Sector Extension and Upgrade for existing sectors. 

Reviewer: ```suggestion
This policy would apply at Sector Extension and [Snap Deal for existing CC sectors](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0019.md). 
```

suggested edit cuz cc upgrade has been replaced by snap deals since nv 15

### Extensions for FIL+ and CC Sectors 

> A point raised by the community concerns the relative ability for Committed Capacity (CC) sectors and FIL+ sectors to extend their sectors upon this policy's implementation. In particular, many of these concerns centered on the ability for CC sectors to instantly extend to gain a potential maximum multiplier on their QAP. FIP-0036 now introduces a softer version of the policy, **capping the max sector duration commitment to 3.5 years** instead of the previously proposed 5-year time-span, and **gradually phasing in the increase to maximum commitment.**

Reviewer: ```suggestion
A point raised by the community concerns the relative ability for Committed Capacity (CC) sectors and sectors with Fil+ deals(verified storage deals) to extend their sectors upon this policy's implementation. In particular, many of these concerns centered on the ability for CC sectors to instantly extend to gain a potential maximum multiplier on their QAP. FIP-0036 now introduces a softer version of the policy, **capping the max sector duration commitment to 3.5 years** instead of the previously proposed 5-year time-span, and **gradually phasing in the increase to maximum commitment.**
```

> This difference in ability for FIL+ and CC sectors to extend was primarily a concern because of potential drops in block reward shares for specific SP profiles. An important note here: drops in block reward shares can always happen. Other SPs can always increase onboarding causing another group’s proportion to drop. The difference then primarily arises from the comparative difficulty in terms of time and effort of onboarding FIL+ compared to CC onboarding or extension. This is concretely mitigated by pledge per QAP dynamics. Though committing soon is good, waiting longer means entering when pledge per QAP is lower.

Reviewer: ```suggestion
This difference in the ability for sectors with Fil+ storage deals and CC sectors to extend was primarily a concern because of potential drops in block reward shares for specific SP profiles. An important note here: drops in block reward shares can always happen. Other SPs can always increase onboarding causing another group’s proportion to drop. The difference then primarily arises from the comparative difficulty in terms of time and effort of onboarding Fil+ compared to CC onboarding or extension. This is concretely mitigated by pledge per QAP dynamics. Though committing soon is good, waiting longer means entering when the pledge per QAP is lower.
```

To make it clear that the inequality of access is not what’s limiting the network, consider an example:
- Two SPs, A and B, each have 1PiB QAP today, 100TiB/day sealing throughput, and 10PiB worth of collateral. 
- SP A can extend CC sectors, and take 3y extension with 3x multiplier. They can quickly gain 3PiB QAP through extension, and a further 7PiB through onboarding over 70days. 

> - SP B has FIL+ deals. They can’t extend. But they can onboard new 3y deals at a rate of up to 3 * 10 * 100TiB/day = 3PiB of QAP/day.

Reviewer: ```suggestion
- SP B has Fil+ deals. They can’t extend. But they can onboard new 3y deals at a rate of up to 3 * 10 * 100TiB/day = 3PiB of QAP/day.
```

this is a bit vague given technically SPs can still extend the sectors with verified deals.

> - FIL+ ability to gain a multiplier on extending is not the limiting factor.

Reviewer: ```suggestion
- Fil+ ability to gain a multiplier on extending is not the limiting factor.
```

> Differences between FIL+ and CC sectors are marginal from an equity of opportunity standpoint, especially given the phasing in of the policy, the cap on maximum duration of 3.5 years, and the multiplicative nature of the Duration incentive on FIL+ ones. The primary limiting factor is new onboarding which is limited by availability of capital flowing into the network. This is a limitation that extending sectors and onboarding of new ones both share in common. Therefore, we shouldn’t strongly weight differences in capacity to extend to be a sufficient source of inequality to reject the proposal. It is more important to promote a stable, long-term oriented network capable of attracting new capital investment. 

Reviewer: ```suggestion
Differences between Fil+ and CC sectors are marginal from an equity of opportunity standpoint, especially given the phasing in of the policy, the cap on the maximum duration of 3.5 years, and the multiplicative nature of the Duration incentive on Fil+ ones. The primary limiting factor is new onboarding which is limited by the availability of capital flowing into the network. This is a limitation that extending sectors and onboarding new ones both share in common. Therefore, we shouldn’t strongly weigh differences in capacity to extend to be a sufficient source of inequality to reject the proposal. It is more important to promote a stable, long-term oriented network capable of attracting new capital investment. 
```

## Test Cases
N/A

## Security Considerations

### Risks of Faulty Proof-of-Replication (PoRep)

> The existing 1.5 year sector duration limit in effect provides a built-in rotation mechanism that can be used to turn over power in the event we discover a flaw in PoRep. Increasing the maximum commitment to 3.5 years weakens this mechanism. [FIP-0047](https://github.com/filecoin-project/FIPs/blob/e498934732765161bbfbafad2a6303dbb7152592/FIPS/fip-0047.md) sets forth a policy to be adopted in case a flaw is discovered in the theory or implementation of proof-of-replication (note, that at this time we are **not** aware of any PoRep issues - FIP-0047 is purely proactive in nature) The policy is that, in case of a PoRep flaw, the RefreshProofExpiration method is disallowed for sectors sealed with the vulnerable code. In order to maintain power, a provider must seal a new replacement sector before each existing sector’s ProofExpriration epoch is reached. If not replaced, a sector with a proof expiration before its commitment expiration will incur a termination fee (for example, the current sector early-termination penalty).

Reviewer: ```suggestion
The existing 1.5 year sector duration limit in effect provides a built-in rotation mechanism that can be used to turn over power in the event we discover a flaw in PoRep. Increasing the maximum commitment to 3.5 years weakens this mechanism. [FIP-0047](https://github.com/filecoin-project/FIPs/blob/e498934732765161bbfbafad2a6303dbb7152592/FIPS/fip-0047.md) sets forth a policy to be adopted in case a flaw is discovered in the theory or implementation of proof-of-replication (note, that at this time we are **not** aware of any PoRep issues - FIP-0047 is purely proactive in nature) The policy is that, in case of a PoRep flaw, the `RefreshProofExpiration` method is disallowed for sectors sealed with the vulnerable code. In order to maintain power, a provider must seal a new replacement sector before each existing sector’s _ProofExpriration_ epoch is reached. If not replaced, a sector with a proof expiration before its commitment expiration will incur a termination fee (for example, the current sector early-termination penalty).
```

are we considering FIP-0047 a prerequiest of this FIP?

### Risks to Consensus
The proposed rewards multiplier increases potential risk to consensus. The main consideration is how long it would take for a colluding consortium of Storage Providers to exceed threshold values of consensus power.

Analysis indicates a malicious consortium would need consistent access to high levels of FIL+, and near-exclusive access to the maximum rewards multiplier, for a substantial period of time, for a viable attack. It is worth noting therefore, that this is the scenario outlined below is an improbable "worst-case" scenario, relying on several unlikely "worst-case" events over sustained periods of time. 

*Example:*
The network currently has 18 EiB of quality adjusted power.  
Consider the scenario of 50 PiB/day onboarding, with 5% attributed to FIL+, and that this is sustained for several months. 

> Now if the malicious consortium can acquire 50% of FIL+ deals and commit sectors for 3.5 years to gain the maximum duration multiplier, and all other storage power maintain the lowest possible duration sectors of 1 year, then in a single day, the adversarial colluding group is expected to gain 0.3% of consensus power. This follows from:

Reviewer: ```suggestion
Now if the malicious consortium can acquire 50% of Fil+ deals and commit sectors for 3.5 years to gain the maximum duration multiplier, and all other storage power maintains the lowest possible duration sectors of 1 year, then in a single day, the adversarial colluding group is expected to gain 0.3% of consensus power. This follows from:
```
